packages/snaps-sdk/src/types/handlers/assets-lookup.ts

```diff
@@ -0,0 +1,75 @@
+import type { Infer } from '@metamask/superstruct';
+import {
+  array,
+  size,
+  literal,
+  number,
+  object,
+  refine,
+  string,
+  record,
+} from '@metamask/superstruct';
+import {
+  assert,
+  CaipAssetTypeStruct,
+  type CaipAssetType,
+} from '@metamask/utils';
+
+export const FungibleAssetUnitStruct = object({
+  name: string(),
+  symbol: string(),
+  decimals: number(),
+});
+
+export type FungibleAssetUnit = Infer<typeof FungibleAssetUnitStruct>;
+
+export const AssetIconUrlStruct = refine(string(), 'Asset URL', (value) => {
+  try {
+    const url = new URL(value);
+    // For now, we require asset URLs to either be base64 SVGs or remote HTTPS URLs
+    assert(
+      url.protocol === 'https:' ||
+        value.startsWith('data:image/svg+xml;base64,'),
+    );
+    return true;
+  } catch {
+    return 'Invalid URL';
+  }
+});
+
+export const FungibleAssetMetadataStruct = object({
+  name: string(),
+  symbol: string(),
+  fungible: literal(true),
+  iconUrl: AssetIconUrlStruct,
+  units: size(array(FungibleAssetUnitStruct), 1, Infinity),
+});
+
+export const OnAssetsLookupResponseStruct = object({
+  assets: record(CaipAssetTypeStruct, FungibleAssetMetadataStruct),
```

Comment:
Should we create an union called `AssetMetadata` that would contain all the asset types in the future (fungible, non-fungible). Right now it would only contain `FungibleAssetMetadata`

Reply:
IMO we can easily replace it with an enum when needed. Not sure we need an union of one element beforehand?

Reply:
Sounds good to me